src/app/debug/debug-tree/debug-tree.component.ts

```diff
@@ -129,94 +119,74 @@ export class DebugTreeComponent implements AfterViewInit, OnDestroy {
 
   addReportToTree(report: Report): void {
     this.lastReport = report;
-    let tree = this.helperService.convertReportToJqxTree(report);
     if (!this.showMultipleAtATime) {
-      this.treeReference.clear();
+      this.tree.clearItems();
     }
-
-    this.treeReference.addTo(tree, null);
-    this.treeReference.selectItem(
-      // @ts-ignore
-      this.treeReference.getItems()[
-        this.treeReference.getItems().findIndex((item: any) => item.id == tree.items[0].id)
-      ],
-    );
+    const newReport: CreateTreeItem = this.transformReportToHierarchyStructure(report);
+    const optional: OptionalParameters = { childrenKey: 'checkpoints', pathAttribute: 'uid' };
+    const path: string = this.tree.addItem(newReport, optional);
+    this.tree.selectItem(path);
     this.hideOrShowCheckpointsBasedOnView(this.currentView);
   }
 
-  selectReport(event: any): void {
-    this.selectReportEvent.emit(event);
-  }
-
-  removeReport(report: any): void {
-    let parentItem: any = this.findParentNode(this.treeReference.getItems().find((item: any) => item.value == report));
-    let root = parentItem.element.parentNode;
-    this.treeReference.removeItem(parentItem);
-    parentItem.element.remove();
-
-    let latestAddedReport = root.lastChild;
-    if (latestAddedReport) {
-      this.treeReference.selectItem(latestAddedReport.querySelectorAll('li')[0]);
+  //Ladybug reports don't have a parent-child structure for its checkpoints, this function creates that parent-child structure
+  transformReportToHierarchyStructure(report: Report): Report {
+    const checkpoints = report.checkpoints;
+    for (let i = 0; i < checkpoints.length; i++) {
```

Comment:
Can't you iterate over checkpoints and use a for...of for this?

Reply:
Items get removed, and the index is being manipulated. This could be done with a filter or a reduce, but it will be more complex, keeping track of multiple arrays.
https://stackoverflow.com/questions/57753537/remove-array-item-using-for-of-loop